semgrep_output_v1.atd

```diff
@@ -441,6 +448,13 @@ type core_match_results
   stats: core_stats;
   (* LATER: rename timing *)
   ?time: core_timing option;
+
+  (* These rules, classified by engine used, will let us be transparent in the CLI
+     output over what rules were run with what.
+   *)
+  rules: rule list;
```

Comment:
maybe an assoc from engine_kind to set of rules would be better than a flat list of rule_id_and_engine_kind?

Can the same  rule be run both with OSS and Pro?

Reply:
I thought about an `(engine_kind * rule_id list) list`, but I was afraid of the fact that, statically, you couldn't ensure that `engine_kind` appeared only once in the `assoc`. I'm fairly sure that a rule can only be run with one engine at a time, and I don't foresee that changing, so I went with the flat list. I can change it though, if you think it's better.

Reply:
In theory we could use (rule_id * engine_kind) list <json repr = "object">
which would make clearer that rule_id is the key and has one engine_kind,
but I guess rule_id_with_engine_kind list is fine too, but could be nice to add in comment
the invariant that's it's really a Map from rule_id to one engine_kind.

Reply:
OK, renamed